packages/components/RadioGroup/src/legacy/RadioButton.win32.tsx

```diff
@@ -0,0 +1,217 @@
+      content,
+      buttonKey,
+      disabled,
+      accessibilityLabel,
+      ariaLabel,
+      componentRef = defaultComponentRef,
+      accessibilityPositionInSet,
+      ariaPosInSet,
+      accessibilitySetSize,
+      ariaSetSize,
+      ...rest
+    } = userProps;
+
+    // Grabs the context information from RadioGroup (currently selected button and client's onChange callback)
+    const info = React.useContext(RadioGroupContext);
+
+    const buttonRef = useViewCommandFocus(componentRef);
+
+    /* We don't want to call the user's onChange multiple times on the same selection. */
+    const changeSelection = () => {
+      if (buttonKey != info.selectedKey) {
+        info.onChange && info.onChange(buttonKey);
+        info.updateSelectedButtonRef && componentRef && info.updateSelectedButtonRef(componentRef);
+      }
+    };
+
+    /* We use the componentRef of the currently selected button to maintain the default tabbable
+    element in a RadioGroup. Since the componentRef isn't generated until after initial render,
+    we must update it once here. */
+    React.useEffect(() => {
+      if (buttonKey == info.selectedKey) {
+        info.updateSelectedButtonRef && componentRef && info.updateSelectedButtonRef(componentRef);
+      }
+    }, []);
+
+    // Explicitly only run on mount and unmount
+    React.useEffect(() => {
+      info.addRadioButtonKey(buttonKey);
+
+      if (!disabled) {
+        info.addRadioButtonEnabledKey(buttonKey);
+      }
+
+      return () => {
+        info.removeRadioButtonKey(buttonKey);
+        info.removeRadioButtonEnabledKey(buttonKey);
+      };
+    }, []);
+
+    const isRTL = I18nManager.isRTL;
+
+    const onInvoke = React.useCallback(
+      (e: KeyPressEvent) => {
+        if (e.nativeEvent.key in DirectionalArrowKeys) {
+          const length = info.enabledButtonKeys.length;
+          const previous =
+            e.nativeEvent.key === DirectionalArrowKeys.ArrowUp ||
+            (isRTL ? e.nativeEvent.key === DirectionalArrowKeys.ArrowRight : e.nativeEvent.key === DirectionalArrowKeys.ArrowLeft);
+          const next =
+            e.nativeEvent.key === DirectionalArrowKeys.ArrowDown ||
+            (isRTL ? e.nativeEvent.key === DirectionalArrowKeys.ArrowLeft : e.nativeEvent.key === DirectionalArrowKeys.ArrowRight);
+          const currRadioButtonIndex = info.enabledButtonKeys.indexOf(info.selectedKey);
+          let newCurrRadioButtonIndex;
+          if (next) {
+            newCurrRadioButtonIndex = (currRadioButtonIndex + 1) % length;
+            info.onChange && info.onChange(info.enabledButtonKeys[newCurrRadioButtonIndex]);
+            info.updateInvoked && info.updateInvoked(true);
+          } else if (previous) {
+            newCurrRadioButtonIndex = (currRadioButtonIndex - 1 + length) % length;
+            info.onChange && info.onChange(info.enabledButtonKeys[newCurrRadioButtonIndex]);
+            info.updateInvoked && info.updateInvoked(true);
+          }
```

Comment:
Is there any case where both next and previous are false? If not, making this an if/else instead could be better so that we make sure that the code falls into one case or the other.

Also the last two lines of each block are identical, we could pull them out.